float32 theta
float32 distance
float32[512] sensors
bool stop
---
int64 success
float32 robot_x
float32 robot_y
float32 theta
